File: 777-toeplitz-matrix/toeplitz-matrix.py
class Solution:
    def isToeplitzMatrix(self, matrix: List[List[int]]) -> bool:
        rows = len(matrix)
        columns = len(matrix[0])

        def is_diagonal_univalue(i, j):
            val = matrix[i][j]
            while (i < rows and j < columns):
                if (matrix[i][j] != val):
                    return False
                i += 1
                j += 1
            
            return True

        for j in range(columns):
            if (not is_diagonal_univalue(0, j)):
                return False
        
        for i in range(1, rows):
            if (not is_diagonal_univalue(i, 0)):
                return False

        return True

File: 777-toeplitz-matrix/toeplitz-matrix.js
var isToeplitzMatrix = function(matrix) {
    let rows = matrix.length
    let columns = matrix[0].length

    function isDiagonalUnivalue(i, j) {
        let cell = matrix[i][j]
        row = i
        column = j
        while (row < rows && column < columns) {
            if (matrix[row][column] != cell) return false
            row += 1
            column += 1
        }
        return true
    }    

    for (let j = 0; j < columns; j++) {
        if (!isDiagonalUnivalue(0, j)) return false
    }

    for (let i = 1; i < rows; i++) {
        if (!isDiagonalUnivalue(i, 0)) return false
    }

    return true
};